spec/ddtrace/configuration/agent_settings_resolver_spec.rb

```diff
@@ -123,9 +168,11 @@
         end
 
         it 'falls back to the defaults' do
-          expect(resolver).to have_attributes default_settings
+          expect(resolver).to have_attributes settings
         end
       end
```

Comment:
Minor: we could add a test here that checked that the uds socket was still used if it exists, since in this branch the custom config was invalid

Reply:
The unix socket won't be used here, as `DD_TRACE_AGENT_PORT` was explicitly provided. We don't want to override use configuration, even if invalid: we want to have the tracer behave as instructed given the provided settings. Of course in this case an `InvalidArgument` will be raised, but that will give the user a clear message on what's the issue in their configuration.